src/coreclr/System.Private.CoreLib/src/System/TypedReference.CoreCLR.cs

```diff
@@ -16,6 +16,17 @@ public ref partial struct TypedReference
         private readonly ref byte _value;
         private readonly IntPtr _type;
 
+        // implementation of CORINFO_HELP_GETREFANY
+        internal static ref byte GetRefAny(IntPtr clsHnd, TypedReference typedByRef)
+        {
+            if (clsHnd != typedByRef._type)
+            {
+                throw new InvalidCastException();
```

Comment:
Your original version has this in throwhelper. Is main path the same with/without throwhelper?

Reply:
I'm testing its performance. It was 0.34 with original change. Now it's 1.34 without the helper. I'm running EgorBot again with 3015a55 to see if it goes back. All measurements are reported at: https://github.com/EgorBot/runtime-utils/issues/200

Reply:
0.58 now https://github.com/EgorBot/runtime-utils/issues/200#issuecomment-2515710871